integrations/slack/client.go

```diff
@@ -34,230 +37,257 @@ var desc = kittehs.Must1(sdktypes.StrictIntegrationFromProto(&sdktypes.Integrati
-			),
+// connStatus is an optional connection status check provided by the
+// integration to AutoKitteh. The possible results are "init required"
+// (the connection is not usable yet) and "using X" (where "X" is the
+// authentication method: an OAuth v2 app, or a Socket Mode app).
+func connStatus(vs sdkservices.Vars) sdkintegrations.OptFn {
+	return sdkintegrations.WithConnectionStatus(func(ctx context.Context, cid sdktypes.ConnectionID) (sdktypes.Status, error) {
+		if !cid.IsValid() {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "init required"), nil
+		}
 
-			// Bots.
-			sdkmodule.ExportFunction(
-				"bots_info",
-				botsAPI.Info,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/bots.info"),
-				sdkmodule.WithArgs("bot", "team_id?"),
-			),
+		vs, err := vs.Get(ctx, sdktypes.NewVarScopeID(cid))
+		if err != nil {
+			return sdktypes.InvalidStatus, err
+		}
 
-			// Chat.
-			sdkmodule.ExportFunction(
-				"chat_delete",
-				chatAPI.Delete,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/chat.delete"),
-				sdkmodule.WithArgs("channel", "ts"),
-			),
-			sdkmodule.ExportFunction(
-				"chat_get_permalink",
-				chatAPI.GetPermalink,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/chat.getPermalink"),
-				sdkmodule.WithArgs("channel", "message_ts"),
-			),
-			sdkmodule.ExportFunction(
-				"chat_post_ephemeral",
-				chatAPI.PostEphemeral,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/chat.postEphemeral"),
-				sdkmodule.WithArgs("channel", "user", "text", "blocks?", "thread_ts?"),
-			),
-			sdkmodule.ExportFunction(
-				"chat_post_message",
-				chatAPI.PostMessage,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/chat.postMessage"),
-				sdkmodule.WithArgs("channel", "text?", "blocks?", "thread_ts?", "reply_broadcast?", "username?", "icon_url?"),
-			),
-			sdkmodule.ExportFunction(
-				"chat_update",
-				chatAPI.Update,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/chat.update"),
-				sdkmodule.WithArgs("channel", "ts", "text?", "blocks?", "reply_broadcast?"),
-			),
+		if vs.Has(vars.AppTokenName) {
+			return sdktypes.NewStatus(sdktypes.StatusCodeOK, "using Socket Mode app"), nil
+		}
+		return sdktypes.NewStatus(sdktypes.StatusCodeOK, "using OAuth v2 app"), nil
+	})
+}
 
-			// Convenience wrappers for "chat.postMessage".
-			sdkmodule.ExportFunction(
-				"send_text_message",
-				chatAPI.SendTextMessage,
-				sdkmodule.WithFuncDoc("convenience wrapper for chat.postMessage"),
-				sdkmodule.WithArgs("target", "text", "thread_ts?", "reply_broadcast?"),
-			),
-			sdkmodule.ExportFunction(
-				"send_approval_message",
-				chatAPI.SendApprovalMessage,
-				sdkmodule.WithFuncDoc("convenience wrapper for chat.postMessage"),
-				sdkmodule.WithArgs("target", "header", "message", "green_button?", "red_button?", "thread_ts?", "reply_broadcast?"),
-			),
+func funcs(vs sdkservices.Vars) []sdkmodule.Optfn {
```

Comment:
I don't like this name. Maybe `initExports` or `exportFuncs`?

Reply:
Renamed to `exportFuncs` in both integrations